plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2926,9 +2926,12 @@ public ConnectorTableHandle makeCompatiblePartitioning(ConnectorSession session,
     @VisibleForTesting
     static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
     {
-        if (partitions.isEmpty()) {
+        if (partitionColumns.isEmpty() && partitions.isEmpty()) {
```

Comment:
```suggestion
        if (partitions.isEmpty()){
            return partitionColumns.isEmpty() ? TupleDomain.none() : TupleDomain.all();
        }
```

Reply:
@albericgenius  `partitions.isEmpty()` check is common in both `if` statements.
Please consider applying the suggested change.

Reply:
Thanks, I can see the improvement from your comments.